src/hook/useInputComplete.tsx

```diff
@@ -0,0 +1,15 @@
+import { useState } from "react";
+
+export const useInputCompleted = () => {
+  const [isCompleted, setIsCompleted] = useState(true);
+
+  const checkInputCompleted = (value: string, length: number) => {
+    setIsCompleted(false);
+
+    if (value.length === length) {
+      setIsCompleted(true);
+    }
+  };
+
+  return { isCompleted, checkInputCompleted };
+};
```

Comment:
이거는.. 상태를 사용하기 위해 억지로 끼워맞춘 느낌이 듭니다. 단순 유효성 체크라 함수 하나면 되는 것을 상태를 두고 함수를 제공해 상태를 바꾸는 것을 통해 컴포넌틑 리렌더링만 추가되었네요! 

```suggestion
const isInputCompleted = ((value: string, length: number) => value.lengh === length;
```

추가로 React 의 Concurrent 에 대해서 공부해보시면 좋을 것 같아요! 헤다가 구현하신 checkInputCompleted 에서 렌더링이 몇번 일어날까요?